src/components/Popup/Popup.tsx

```diff
@@ -28,6 +36,7 @@ export interface IPopupProps {
 
 export interface IPopupState {
   triggerRef: HTMLElement
+  popupOpened?: boolean
```

Comment:
Why is `Popup` not an auto controlled component (with the same prop name as `Portal`)?

Reply:
Actually, a good question :) Initial implementation of Popup doesn't expose this prop to users, and when I was making this change, I didn't want to change API of a component but leave as it is. 
So only behavior and key handlers would be added. Anyway, this can be a point for discussion as well as a separate PR.

Reply:
yes, lets address it by means of the follow up PR, will take care of that